### PR TITLE
Add Cert Manager

### PR DESCRIPTION
**Pull request by timaa2k**

### CODE REVIEW

**Comment by hectorj2f** [comment]
@timaa2k please, sign your commits :)

**Comment by shaneutt** [comment]
I'm excited to see cert-manager added :+1: 

While this is in `wip` and you're working on this, please feel free to check in with us in the #kubeaddons channel on Slack if you need any help with anything! (Or pinging here too is :+1: )

**Comment by hectorj2f** [comment]
@timaa2k We need to add that functionality to the chart to create the cluster-issuer resource once the cert-manager service is responding.

File: templates/cert-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  name: cert-manager
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: cert-manager
+  annotations:
+    appversion.kubeaddons.mesosphere.io/cert-manager: "0.1.0"
+    docs.kubeaddons.mesosphere.io/cert-manager: "https://docs.cert-manager.io/en/release-0.10/"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.7.0
```
Comment (shaneutt) [inline templates/cert-manager.yaml]: while the _chart_ may support this low of a version Addons in kubeaddons general take the stance of v1.15+ as per [kubeaddons requirements](https://github.com/mesosphere/kubeaddons#requirements) and just out of general caution. Any reason we shouldn't update this?
```suggestion
    minSupportedVersion: v1.15.0
```

File: templates/cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  cloudProvider:
+    - name: aws
+      enabled: true
+    - name: docker
```
Comment (shaneutt) [inline templates/cert-manager.yaml]: Is there anything special in a docker environment for this, or does it `justwork.py`?

**Review by shaneutt — CHANGES_REQUESTED** [review]

File: templates/cert-manager.yaml (hunk 2 above)
Comment (timaa2k) [reply templates/cert-manager.yaml]: It `just works (TM)` :)

**Review by shaneutt — APPROVED** [review]
:heavy_check_mark: once we're unblocked on https://github.com/mesosphere/charts/pull/123

:+1:

**Review by hectorj2f — CHANGES_REQUESTED** [review]
`spec.namespace` needs to be set to `cert-manager` as done for `velero` here https://github.com/mesosphere/kubeaddons-configs/blob/master/templates/velero.yaml#L34

**Comment by jimmidyson** [comment]
@hectorj2f Why does cert-manager need to be in it's own namespace here?

**Comment by timaa2k** [comment]
We upload the `kubernetes-root-ca` to a separate namespace `cert-manager`, so that only `cert-manager` can see it. The issuer it sets up initially also needs to only be accessible in that namespace.

**Comment by joejulian** [comment]
We don't with other Addons, but should we start prefixing namespaces to more clearly designate ownership? `kubeaddons-cert-manager`?

File: templates/dex-k8s-authenticator.yaml [hunk 3]
```diff
@@ -55,13 +55,7 @@ spec:
...
-      caCerts:
-        enabled: true
-        secrets:
-        - name: dex-root-ca
-          filename: dex-root-ca.crt
-          value: cGxhY2Vob2xkZXIK # b64 "placeholder" this will be updated by initcontainers
       initContainers:
       - name: initialize-dka-config
-        image: mesosphere/kubeaddons-addon-initializer:v0.0.4
+        image: timaa2k/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/dex-k8s-authenticator.yaml]: latest ?

Comment (hectorj2f) [reply templates/dex-k8s-authenticator.yaml]: We decided to go with version tags :/

File: templates/dex.yaml [hunk 4]
```diff
@@ -22,7 +22,7 @@ spec:
       kubeaddons.mesosphere.io/provides: ingresscontroller
   chartReference:
     chart: dex
-    repo: https://mesosphere.github.io/charts/stable
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/dex.yaml]: let's use `https://mesosphere.github.io/charts/stable` and bump the dex chart version

File: templates/traefik-forward-auth.yaml [hunk 5]
```diff
@@ -11,7 +11,7 @@ spec:
       kubeaddons.mesosphere.io/name: dex    
   chartReference:
     chart: traefik-forward-auth
-    repo: https://mesosphere.github.io/charts/staging
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/traefik-forward-auth.yaml]: https://mesosphere.github.io/charts/staging and back the version accordingly

File: templates/traefik.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ spec:
       kubeaddons.mesosphere.io/provides: defaultstorageclass
   chartReference:
     chart: traefik
-    repo: https://mesosphere.github.io/charts/staging
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/traefik.yaml]: let's revert https://mesosphere.github.io/charts/staging and bump to the right chart version

**Review by hectorj2f — CHANGES_REQUESTED** [review]
We need to bump dex, traefik and trafik-auth to the chart versions and revert the repos to mesosphere/charts/stable or staging

**Comment by shaneutt** [comment]
Seems like we still have some work in progress on this one @timaa2k & @hectorj2f? Let me know if any assistance could be useful to get unblocked and/or finish (code, testing, docs, e.t.c.)

**Review by hectorj2f — APPROVED** [review]

**Comment by hectorj2f** [comment]
This branch got deleted and our Konvoy PR failed because this branch doesn't exist anymore :(. We will point to master now

**Pull request merged**